src/org/project/Cart.java

```diff
@@ -0,0 +1,80 @@
+package org.project;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cart {
+    private List<Product> cartItems;
+
+    public Cart() {
+        this.cartItems = new ArrayList<>();
+    }
+
+    public void addProductToCart(Product product) {
+        if (product == null) {
+            throw new IllegalArgumentException("Product cannot be null.");
+        }
+        if (product.getQuantity() <= 0) {
+            throw new IllegalArgumentException("No pieces of product available: " + product.getName());
+        }
+
+        for (Product cartProduct : cartItems) {
```

Comment:
to zadziala okej, ale z perspektywy clean codu powinnismy taki fragment przeniesc do innej meotdy, nazwac go w stylu isProductInCart i na podstawie wyniku takiej meotdy decydowac czy zwiekszamy quanitity czy dodajemy produkt do listy nowy bo go nie ma, taka petla ktora ma w sobie ifa na przypadek ktory obslugujemy w dzialaniu jest spoko ale nie wyglada profesjonalnie i zaciemnia kod